src/encryption.rs

```diff
@@ -0,0 +1,123 @@
+use crate::params::{PvwError, PvwParameters, Result};
+use crate::public_key::GlobalPublicKey;
+use fhe_math::rq::{Context, Poly, Representation};
+use fhe_util::sample_vec_cbd;
+use num_bigint::{BigUint, ToBigUint};
+use num_traits::ToPrimitive;
+use rand::{CryptoRng, RngCore};
+use std::sync::Arc;
+
+/// Ciphertext output of PVW encryption
+#[derive(Debug, Clone)]
+pub struct PvwCiphertext {
+    /// c1 = A * r^T + e1'  ∈ ℤ_q^k
+    pub c1: Vec<Poly>,
+    /// c2 = b * r^T + e2' + c  ∈ ℤ_q
+    pub c2: Vec<Poly>,
+}
+
+pub fn encrypt<R: RngCore + CryptoRng>(
+    params: &PvwParameters,
+    rng: &mut R,
+    ctx: &Arc<Context>,
+    pk: &GlobalPublicKey,
+    message: &[u8],
```

Comment:
thinking out loud but this should be `u64` for me since our generate secret shares [fn in trbfv](https://github.com/gnosisguild/fhe.rs/blob/67b4bda0cbd48ff91188ce611107212399f31708/crates/fhe/src/trbfv/shares.rs#L141) returns u64.

Reply:
This makes more sense
I followed a bit pattern kind of approach, but this makes more sense